### DISCUSSION TITLE
Genetics and why they're not good (unstable update)

### DISCUSSION THREAD

**Discussion by ubertpendragon** [FakeFishGames/Barotrauma#14296] [Unstable feedback]
Since Genetics are getting a mild rework, this is what feels wrong about the current implementation:

**1. Tainted means Trash**
Tainted genes cannot be combined, refined or reused in any way. They are forever locked to a bad condition%, as well as having a debilitating downside, _specifically Hypersensitivity and Rigid Joints._ You want to AVOID TAINTING instead of dealing with the side effect (see below)

**2. Too talent reliant**
Since you want to avoid tainting at all costs, **you absolutely want the medic talents before engaging in genetics.** 
For REFINING, you want `Gene Splicer` to increase the efficiency of combining by a factor of 3x to 2.25x
For COMBINING, you want `Genetic Genius` to eliminate the possibility of tainting entirely.
**You're incentivised to hoard genes until the talents kick in. You're incentivized to only take the gene related talents** in the Xenobiologist tree.

**3. Impossible to remove genes**
When you remove an equipped splicer, they delete genes inside. This means you do not want to equip low quality genes until you can refine them later. **You're incentivized to hoard genes instead of equipping them.**

**4. Bad genes/No scaling**
Some genes are bad because they don't scale properly. Spineling/Mudraptor/Husk. Some miss a "step" at 100% condition (Hunter/Mollusk/Skitter)

**5. Getting genes is boring/random**
Simply put, it's boring. Either loot every corpse or buy a ton of them at every merchant you encounter. And sometimes you get hammerhead genes from dead crawlers which is just weird.

**6. hideconditionbar="true"**
Don't hide it, it makes it harder to tell gene quality apart when opening a cabinet.

-------------

# Proposed changes:

* ENFORCE TAINTING, rebalance tainted effects.
Controversial, but it's more interesting having to deal with negative effects than outright ignoring them and tossing any tainted genes in the trash. Some tainted effects can be rebalanced to not be as detrimental.

* Allow tainted genes to be refined/combined/recycled
Tainted genes are hardcoded to not be reused. This prevents reusing them.  [Example of trying to reuse tainted genes by making temporary "potions" out of them](https://cdn.discordapp.com/attachments/668819422919524352/1225175179714363452/Desktop_2023.09.10_-_17.38.09.06_38.mp4?ex=66962a8d&is=6694d90d&hm=8b856cdccc26a8d27d7605a800c8a94dbc1bb9adb9f19a6f5de800bfb8166bfd&)

* Do not delete genes when removed. 
Commit says _Removing the splicer degrades the genetic material instead of destroying it_, which is good.

* Rebalance talents and refine formula
For Gene Splicer, change the refining formula
`highest quality input + [5-8]%, OR highest quality input + [15-18]% (Gene Splicer)`
Increase the base 5-8% to 10-13, and reduce gene splicer increase to only 5% (overall buff).
For Genetic Genius, _anything that isn't removing tainting from the game._  Could be a vitality bonus when equipping tainted genes for example

* Accept my spineling genes PR :^)

* A new gene extraction system
Extract genes from LIVE creatures. Get genes matching the creature species. Rebalance genes so that the stronger species have the stronger gene effects.

https://github.com/user-attachments/assets/8e7d13be-20ed-4d15-af5d-372bb3865d04

-----

Also see #13922 #12902 #12383

**Comment by WJohn** [comment]
I'm personally in the camp that the entire condition system for genes is inherently bad. It promotes holding onto genes until they reach 100% -- effectively meaning nobody gets genes because no one wants an 11% skitter gene. I would much rather just have more actually different types of genes instead of this scaling system.

The current tainted genes system is purely a punishment system, and like Fallout 3's speech checks, can be easily bypassed through save scum. This is neither an effective punishment, nor is a punishment system even a good idea. Players spend ages even collecting genes, why on earth make the process even more grindy? A system where genes come with positives and negatives no matter what you do is far more interesting.

On the note of collecting genes, that entire system is screwed beyond belief. Your rate of gene collection is agonizingly slow without talents. You rarely ever find any inside alien ruins or shipwrecks. Gene Harvester promotes Captains to stop and collect each and every big swarm -- adding to mission time. And then the Blackmarket Genes talent completely throws all previous methods in the trash by dumping 7-15 genes on you at every Colony and Research Outpost.

Lastly, genes being destroyed or damaged on death adds insult to injury. This is by far the most grindy, minmaxing system in the game where you put all your eggs in one basket -- except it can all be smashed and lost due to someone slapping cyanide on you, or your body being completely devoured by mudraptors. You know what other massive power buffing mechanic exists without any of this grind or threat of losing everything? Endocrine Boosters.

In my opinion, genes should be much more powerful, but also be sort of "disposable" -- swapping them out or losing them should be quick and easy and free of any punishment. The difficulty should come from dealing with their individual tradeoffs and drawbacks, not on having only a single gene splicer ready after 40 hours and sweating over the possibility of losing it.

**Comment by ubertpendragon** [comment]
Feedback from testing the new Unstable genes:

Mostly good changes, especially not ending up with forever useless tainted genes and the refining rescale, BUT...

* **Gene hoarding is still encouraged.** See below

* **Genes losing 10% condition when removed is too much**. 
Major improvement from *destroying* the genes entirely, but -10% quality means losing exactly a 26% gene in the process. The right number is 0%. You take damage when removing genes (not enough?), and dying taints the gene anyway, so there's already a downside to removing genes.

* **Gene splicer is still extremely strong**. 
Just wait for it before doing any gene refining. Before, it increased refining potency to 300-225%, with the new formula it's 300-150%. Average "fresh" gene quality is 26%, which puts it at 200% efficieny; it basically doubles your refining quality. It's also way better than Bloody Business (large enemies give 1 extra bloodbags? small ones always do... less psychosis? useless)

* **Genetic Genius isnt as mandatory** so it's a plus, as you can remove taint by brute forcing combines.


Overall:

* Would still like to see enforced tainting, probably better than bypassing tainting with talents entirely
* Talent rebalance somehow (especially gene splicer vs blood) ?
* Husk, Spineling, Mudraptor need some love
* Show condition bar on genes
* Change tags on research station for modding #13922

**Comment by ubertpendragon** [comment]
Related: 
There's a bug (?) where if you combine a gene with a non-gene to produce a 3rd item that also isnt a gene, the initial gene doesnt get deleted in the process 

Example: gene combined with alien blood, produces "temporary" gene potion, but the genes dont get eaten in the process
![image](https://github.com/user-attachments/assets/1d268a86-6161-446a-8ef4-f1bd2f960e5f)
![image](https://github.com/user-attachments/assets/a0c55a47-bdbb-4b28-8118-37ff555ea2d1)

```xml
      <Item identifier="temporarycrawlergenes" requiredotheritem="alienblood" copycondition="true" activatebuttontext="researchstation.research" infotext="researchstation.research.infotext" infotextonotheritemmissing="researchstation.combine.missingitem" />
 ````

Here's the mod prototype if you want to try
[genetweaks.zip](https://github.com/user-attachments/files/16731244/genetweaks.zip)

**Comment by gerakis** [comment]
I would like to see reusable genetics. Just the fact that you could say "harvest his geneseed" after someone dies would be really funny.
